torch/onnx/utils.py

```diff
@@ -915,6 +914,60 @@ def _decide_input_format(model, args):
     return args
 
 
+@_beartype.beartype
+def _from_dynamic_axes_to_dynamic_shapes(
+    model,
+    dynamic_axes: Optional[
+        Union[Mapping[str, Mapping[int, str]], Mapping[str, Sequence[int]]]
+    ] = None,
+    input_names: Optional[Sequence[str]] = None,
+) -> Optional[Union[Dict[str, Any], Tuple[Any], List[Any]]]:
+    """
+
+    dynamic_axes examples:
+    (1) dynamic_axes = {"x": {0: "my_custom_axis_name_1"}, "y": {1: "my_custom_axis_name_2"}}
+    (2) dynamic_axes = {"x": [0], "y": [1]}
+
+    these will be converted to dynamic_shapes respectively:
+    (1) dynamic_shapes = {"x": {0: Dim("my_custom_axis_name_1")}, "y": {1: Dim("my_custom_axis_name_2")}}
+    (2) dynamic_shapes = {"x": {0: Dim("x_dim_0")}, "y": {1: Dim("y_dim_1")}}  # auto-generated dim names
+
+    """
+    if dynamic_axes is None:
+        return None
+    if input_names is None:
+        input_names = []
+    dynamic_shapes: Dict[str, Optional[Any]] = {}
+    for input_name, axes in dynamic_axes.items():
+        if input_name in input_names:
+            raise ValueError(
+                "input names is not supported yet. Please use model forward signature."
+            )
```

Comment:
Perhaps explain the error a little bit more? Reading it briefly it isn't clear to me what is wrong and what needs to be done

Reply:
For the time being, we only support:

```python

class Model(nn.module):
    def forward(self, input_x, input_y):
        ...

torch.onn.export(Model(), args, dynamic_axes={"input_x": ..., "input_y": ...}, dynamo=True)
```

input_names is not changing anything, so users have to use forward sig. Any suggestions?

Reply:
Got it. We can support that later. Thanks!